src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -1,23 +1,74 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
 using Microsoft.OpenApi.Models;
 
 namespace Microsoft.AspNetCore.OpenApi;
 
-internal sealed class OpenApiDocumentService(IHostEnvironment hostEnvironment)
+internal class OpenApiDocumentService(
+    [ServiceKey] string documentName,
+    IApiDescriptionGroupCollectionProvider apiDescriptionGroupCollectionProvider,
+    IHostEnvironment hostEnvironment,
+    IOptionsSnapshot<OpenApiOptions> optionsSnapshot)
 {
+    private readonly OpenApiOptions _options = optionsSnapshot.Get(documentName);
+
     public Task<OpenApiDocument> GetOpenApiDocumentAsync()
     {
         var document = new OpenApiDocument
         {
-            Info = new OpenApiInfo
-            {
-                Title = hostEnvironment.ApplicationName,
-                Version = OpenApiConstants.DefaultOpenApiVersion
-            }
+            Info = GetOpenApiInfo(),
+            Paths = GetOpenApiPaths()
         };
         return Task.FromResult(document);
     }
+
+    internal OpenApiInfo GetOpenApiInfo()
+    {
+        return new OpenApiInfo
+        {
+            Title = $"{hostEnvironment.ApplicationName} | {documentName}",
```

Comment:
This is only for humans (and AI), right?  It's never parsed to obtain the parts?

Reply:
Correct. The spec doesn't place any requirements on the format of the title.